### PR TITLE
Cordova client bug fixes

### PR DESCRIPTION
**Pull request by shrishrirang**
Closes https://github.com/Azure/azure-mobile-apps-js-client/issues/217 and https://github.com/Azure/azure-mobile-apps-js-client/issues/201

### CODE REVIEW

**Comment by shrishrirang** [comment]
@danderson00

File: sdk/test/tests/target/cordova/pushError.tests.js [hunk 1]
```diff
@@ -177,4 +236,11 @@ function verifyChangeAction(oldAction, newAction, oldItemId, newItemId, oldVersi
         $assert.isNull(batchExecuted);
         $assert.isFalse(isSuccessExpected);
     });
+}
+
+function makeCopy(obj) {
+    if (obj === undefined || obj === null) {
+        return obj;
+    }
+    return JSON.parse( JSON.stringify(obj) );
```
Comment (danderson00) [inline sdk/test/tests/target/cordova/pushError.tests.js]: This is a relatively slow method of cloning. Only worth changing if you're looking for optimisations

**Comment by danderson00** [comment]
:shipit:

File: sdk/test/tests/target/cordova/pushError.tests.js (hunk 1 above)
Comment (shrishrirang) [inline sdk/test/tests/target/cordova/pushError.tests.js]: Yeah this fix is primarily for circular ref issue. Perf optimizations of this scale one of the last concerns right now :)

**Pull request merged**